src/extension/nodes/CompoundCloudPlane.cpp

```diff
@@ -0,0 +1,479 @@
+#include "CompoundCloudPlane.hpp"
+#include <godot_cpp/classes/resource_loader.hpp>
+#include <godot_cpp/classes/shader.hpp>
+#include <godot_cpp/classes/quad_mesh.hpp>
+#include <godot_cpp/classes/fast_noise_lite.hpp>
+#include <godot_cpp/variant/utility_functions.hpp>
+
+namespace Thrive {
+
+std::unordered_map<int64_t, CompoundCloudPlane*> CompoundCloudPlane::instances;
+int64_t CompoundCloudPlane::next_instance_id = 1;
+
+void CompoundCloudPlane::_bind_methods() {
+    using namespace godot;
+
+    ClassDB::bind_method(D_METHOD("get_native_instance"), &CompoundCloudPlane::get_native_instance);

```

Comment:
But when exposed through Godot the godot naming convention should be used. So this line should be:
```suggestion
    ClassDB::bind_method(D_METHOD("get_native_instance"), &CompoundCloudPlane::GetNativeInstance);

```